src/main.cpp

```diff
@@ -3871,13 +3871,16 @@ DEMUX_PACKET* CInputStreamAdaptive::DemuxRead(void)
     else
       p = AllocateDemuxPacket(iSize);
 
-    p->dts = static_cast<double>(sr->DTS() + m_session->GetChapterStartTime());
-    p->pts = static_cast<double>(sr->PTS() + m_session->GetChapterStartTime());
-    p->duration = static_cast<double>(sr->GetDuration());
-    p->iStreamId = sr->GetStreamId();
-    p->iGroupId = 0;
-    p->iSize = iSize;
-    memcpy(p->pData, pData, iSize);
+    if (iSize > 0 && pData)
```

Comment:
Previously the condition was just `if (iSize)` - is the extra condition check on `pData` necessary? Not trying to nitpick, were you getting crashes without it?